src/main/scala/is/hail/HailContext.scala

```diff
@@ -233,14 +233,20 @@ class HailContext private(val sc: SparkContext,
     tolerance: Double = 0.2,
     nPartitions: Option[Int] = None): VariantDataset = {
 
-    val inputs = hadoopConf.globAll(files)
+    val inputs = hadoopConf.globAll(files).flatMap { file =>
+      if (hadoopConf.isDir(file))
+        hadoopConf.glob(file + "/part-*").filter(!_.getPath.getName.endsWith(".idx")).map(_.getPath.toString)
+      else
+        Array(file)
+    }
 
     if (inputs.isEmpty)
       fatal(s"arguments refer to no files: '${ files.mkString(",") }'")
 
     inputs.foreach { input =>
-      if (!input.endsWith(".bgen"))
-        fatal("unknown input file type")
+      if (!input.endsWith(".bgen") && !input.contains("part")) {
```

Comment:
I'd put this above and just warn in terms path specified by the user.  So they should give us a list of paths that end in .bgen, but can be a file or directory.